memecity.engine/Engine/SDL/GraphicsFacade.cpp

```diff
@@ -15,23 +17,20 @@ namespace memecity::engine::sdl {
 	{
 		if (SDL_Init(SDL_INIT_VIDEO) < 0)
 		{
-			printf("SDL initialization error: %s\n", SDL_GetError());
-			return false;
+			throw SDLException(Level::error, SDL_GetError());
 		}
 
 		sdl_window = std::make_unique<RawWindowWrapper>(SDL_CreateWindow("MemeCity", SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED, screen_width, screen_height, SDL_WINDOW_SHOWN));
 		if (sdl_window == nullptr)
 		{
-			printf("Window creation error: %s\n", SDL_GetError());
-			return false;
+			throw SDLException(Level::error, SDL_GetError());
 		}
 
 		if (is_fullscreen)
 		{
 			if (SDL_SetWindowFullscreen(sdl_window->get(), SDL_WINDOW_FULLSCREEN_DESKTOP) != 0)
 			{
-				printf("Window FullScreen error: %s\n", SDL_GetError());
-				return false;
+				throw SDLException(Level::error, SDL_GetError());
```

Comment:
weet je zeker dat je het spel wil laten stoppen bij een sld error voor fullschreen? ipv het spel klein te houden en een waarschuwing door te geven?

Reply:
Als we dat willen, moet dat aan de callende kant worden besloten